docs/rules/no-unused-expressions.md

```diff
@@ -39,6 +39,18 @@ a && b()
 a, b()
 
 c = a, b;
+
+(function namedFunctionInExpressionContext () {f();});
```

Comment:
Do you think of this as a possible mistake when the intent was an immediately-invoked function expression?

If so, might be worth a comment here, or corresponding IIFE in the correct code examples below, or both.